libraries/libfc/include/fc/container/ordered_diff.hpp

```diff
@@ -108,15 +108,17 @@ class ordered_diff {
       // Remove from the source based on diff.remove_indexes
       std::ptrdiff_t offset = 0;
       for (SizeType index : diff.remove_indexes) {
-         FC_ASSERT(index + offset < container.size(), "diff.remove_indexes index ${idx} + offset ${o} not in range ${s}",
+         SizeType updated_index = index + offset;
```

Comment:
This should not be `SizeType` as that is often going to be unsigned. Use `auto` instead.